### PR TITLE
Migrate Stream Wrapper to ArmoniK.Api

### PR DESCRIPTION
**Pull request by aneojgurhem**

### CODE REVIEW

**Review by ngruelaneo — APPROVED** [review]

**Review by lemaitre-aneo — COMMENTED** [review]
Would have been cool to not have formatting changes in this PR... Apart from that, all is good.

**Comment by aneojgurhem** [comment]
The formatting changes come from the fact that those files were modified during this pr

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +0,0 @@
-[submodule "Api"]
-	path = Api
-	url = https://github.com/aneoconsulting/ArmoniK.Api.git
```
Comment (jfonseca-aneo) [inline .gitmodules:3]: the submodule is only required for the StreamWrapper?

Comment (aneojgurhem) [reply .gitmodules:3]: No, but I made packages in ArmoniK.Api. Therefore, I used them instead of the submodule

Comment (jfonseca-aneo) [reply .gitmodules:3]: So if we remove it as submodule in Core we need to document the requirement (in case someone clones the repo)

Comment (aneojgurhem) [reply .gitmodules:3]: The package is included in the .csproj so this makes it a requirement already.

Comment (jfonseca-aneo) [reply .gitmodules:3]: Ah ok!

File: Tests/HtcMock/Client/src/ArmoniK.Samples.HtcMock.Client.csproj [hunk 2]
```diff
@@ -24,16 +24,18 @@
 	</PropertyGroup>
 
 	<ItemGroup>
+	  <PackageReference Include="ArmoniK.Api.Client" Version="3.0.0-jgmigrate.81.e1930f1" />
```
Comment (jfonseca-aneo) [inline Tests/HtcMock/Client/src/ArmoniK.Samples.HtcMock.Client.csproj:27]: we need a proper tag here

**Review by jfonseca-aneo — APPROVED** [review]

**Pull request merged**